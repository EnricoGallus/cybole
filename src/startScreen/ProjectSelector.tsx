import React, {ReactNode} from 'react';
import {useNavigate} from 'react-router-dom';
import {Menu} from "primereact/menu";
import {MenuItemCommandParams} from "primereact/menuitem";

type PropsType = {
    children: ReactNode;
};

const ProjectSelector = (props: PropsType) => {
    const navigation = useNavigate();
    const {children} = props;
    const handleMenu = (route: string) => {
        navigation(route);
    };

    const menuItems = [
        {
            label: 'Menu',
            items: [
                {
                    label: 'Projects',
                    icon: 'pi pi-align-justify',
                    command: () => {
                        handleMenu('/')
                    }
                },
                {
                    label: 'Settings',
                    icon: 'pi pi-cog',
                    command: () => {
                        handleMenu('/settings')
                    }
                }
            ]
        }
    ];

    return (
        <div className="container">
            <div className="grid">
                <div className="card col-12">
                    <h5>Projects</h5>
                </div>
            </div>
            <div className="grid">
                <div className="col-3">
                    <Menu model={menuItems}/>
                </div>
                <div className="col-9">
                    {children}
                </div>
            </div>
        </div>
    );
};

export default ProjectSelector;
