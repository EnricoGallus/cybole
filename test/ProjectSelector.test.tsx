import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProjectSelector from '../src/startScreen/ProjectSelector';

describe('ProjectSelector', () => {
    test('renders component', () => {
        const { container } = render(<ProjectSelector children={<div data-testid="project-open" />} />, {
            wrapper: MemoryRouter,
        });

        expect(container).toMatchSnapshot();
    });
});
