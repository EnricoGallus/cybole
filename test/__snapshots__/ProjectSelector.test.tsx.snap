// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelector renders component 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="card col-12"
    >
      <h5>
        Projects
      </h5>
    </div>
    <div
      class="col-3 pl-5"
    >
      <div
        class="p-menu p-component"
      >
        <ul
          class="p-menu-list p-reset"
          role="menu"
        >
          <li
            class="p-submenu-header"
            role="presentation"
          >
            Menu
          </li>
          <li
            class="p-menuitem"
            role="none"
          >
            <a
              class="p-menuitem-link"
              href="#"
              role="menuitem"
              tabindex="0"
            >
              <span
                class="p-menuitem-icon pi pi-align-justify"
              />
              <span
                class="p-menuitem-text"
              >
                Projects
              </span>
            </a>
          </li>
          <li
            class="p-menuitem"
            role="none"
          >
            <a
              class="p-menuitem-link"
              href="#"
              role="menuitem"
              tabindex="0"
            >
              <span
                class="p-menuitem-icon pi pi-cog"
              />
              <span
                class="p-menuitem-text"
              >
                Settings
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="col-9"
    >
      <div
        data-testid="project-open"
      />
    </div>
  </div>
</div>
`;
